copy/copy.go

```diff
@@ -269,8 +273,11 @@ func Image(ctx context.Context, policyContext *signature.PolicyContext, destRef,
 		if len(options.EnsureCompressionVariantsExist) > 0 {
 			return nil, fmt.Errorf("EnsureCompressionVariantsExist is not implemented when not creating a multi-architecture image")
 		}
+		if (options.DestinationCtx == nil || options.DestinationCtx.CompressionFormat == nil) && options.ForceCompressionFormat {
```

Comment:
(Absolutely non-blocking: Checking `options.ForceCompressionFormat` first, before looking at `DestinationCtx`  might be easier to follow (“if you request to force a compression format, you also must say which one”, rather than “if there is no compression format defined, forcing a compression format is makes no sense”; also all of this code is _primarily_ concerned with `ForceCompressionFormat`, so it makes sense to lead with that).

I suspect with a new helper function the ease of understanding will not be worth worrying about…)